src/videojs-contrib-hls.js

```diff
@@ -78,6 +78,45 @@ const safeGetComputedStyle = function(el, property) {
 };
 
 /**
+ * Updates the selcetedIndex of the QualityLevelList when a mediachange happens in hls.
+ *
+ * @param {QualityLevelList} qualityLevels The QualityLevelList to update.
+ * @param {PlaylistLoader} playlistLoader PlaylistLoader containing the new media info.
+ * @function handleHlsMediaChange
+ */
+const handleHlsMediaChange = function(qualityLevels, playlistLoader) {
+  let newPlaylist = playlistLoader.media();
+  let selectedIndex = -1;
+
+  for (let i = 0; i < qualityLevels.length; i++) {
+    if (qualityLevels[i].id === newPlaylist.uri) {
+      selectedIndex = i;
+      break;
+    }
+  }
+
+  qualityLevels.selectedIndex_ = selectedIndex;
+  qualityLevels.trigger({
```

Comment:
Yes, although selectedIndex shouldn't ever be -1 unless a user manually changes the quality level list, if listening to the change event, this will allow them to see the invalid state and correct or catch any errors instead of this changing out from under them